packages/flutter/lib/logging.dart

```diff
@@ -0,0 +1,159 @@
+///
+/// produces a log event of [Level.INFO].
+///
+/// If no client is subscribed to its corresponding stream, logging calls are
+/// ignored (and no event passed on to the `dart.developer` log function).  The
+/// cost of logging calls can be further mitigated at call sites by invoking
+/// them in a function that is only evaluated in profile or debug modes. For
+/// example,
+///
+/// ```dart
+/// profile(() {
+///   _xLogger.info('X is starting...');
+/// });
+/// ```
+///
+/// In this case, logging is entirely ignored in release mode and no
+/// performance penalty is paid.
+///
+/// Clients interact with the loggers via VM Services.  Specifically,
+///
+///   * 'loggers' : provides an enumeration of available loggers
+///   * 'logger.subscribe' : supports logger subscriptions
+///
+/// See <https://github.com/dart-lang/logging> for more details on
+/// `package:logging`.
+///
+/// See <https://api.dartlang.org/stable/2.0.0/dart-developer/log.html> for more
+/// on the `log` function.
+///
+library logging;
+
+import 'dart:developer' as developer;
+
+import 'package:flutter/src/logging/logging.dart';
+import 'package:logging/logging.dart';
+
+/// A Logger object is used to log messages for specific systems or components.
+class Logger {
+  static int _sequenceNumber = 0;
+
+  /// Logger name.
+  final String name;
+
+  /// Logger description.
+  final String description;
+
+  /// Singleton constructor. Calling `Logger(name)` returns the same
+  /// actual instance whenever it is called with the same string name.
+  factory Logger(String name, {String description}) =>
+      _getOrRegisterLogger(name, description: description ?? '');
+
+  Logger._(this.name, {this.description});
+
+  static Logger _getOrRegisterLogger(String name, {String description}) =>
+      LoggingService.instance.loggers.putIfAbsent(
+          name, () => new Logger._(name, description: description));
+
+  /// The fully qualified name of this logging stream.
+  String get fullName => name;
+
+  /// Log message at level [Level.CONFIG].
+  void config(dynamic message, [Object error, StackTrace stackTrace]) {
+    log(Level.CONFIG, message, error, stackTrace);
+  }
+
+  /// Log message at level [Level.FINE].
+  void fine(dynamic message, [Object error, StackTrace stackTrace]) {
+    log(Level.FINE, message, error, stackTrace);
+  }
+
+  /// Log message at level [Level.FINER].
+  void finer(dynamic message, [Object error, StackTrace stackTrace]) {
+    log(Level.FINER, message, error, stackTrace);
+  }
+
+  /// Log message at level [Level.FINEST].
+  void finest(dynamic message, [Object error, StackTrace stackTrace]) {
+    log(Level.FINER, message, error, stackTrace);
+  }
+
+  /// Log message at level [Level.INFO].
+  void info(dynamic message, [Object error, StackTrace stackTrace]) {
+    log(Level.INFO, message, error, stackTrace);
+  }
+
+  /// Log [message] at the given [level] with optional [error] and [stackTrace].
+  void log(Level level, dynamic message,
```

Comment:
i don't understand the purpose of log, info, finest, finer, fine, and config. None of those seem to be in service of anything in the use cases doc.

Reply:
ditto severe, should, warning, and log

Reply:
You're right.  These are just the levels supported by `package:logging` and familiar to other logging frameworks. 
 They play nice w/ IDEs where you can set filtering on severity.  

![image](https://user-images.githubusercontent.com/67586/45647611-25a34180-ba7b-11e8-8f3d-a3f24a210122.png)

Adding a bunch of helper methods is just convenience but this is assuming there's value...  Do you agree that the notion of severity is useful?

If it's not generally useful, we could leave it up to the specific streams to define a level as part of their message structure (if they'd value it).

Reply:
Nothing in the use case doc motivates any of the clients of `log` as far as I can tell. All the use cases are for passing structure data, none mention severity.

Severity in generally seems useless to me in logging frameworks as a whole. In practice you rarely want "just give me more", you want to turn on specific kinds of logs. But that's academic for this PR which isn't about logging in the first place but about providing data to IDEs for visualisation purposes.